### ISSUE TITLE
Error port data is empty

### ISSUE THREAD

**Issue by chanskw** [IBMStreams/streamsx.jdbc#33]
Unsure if there is something wrong with my app.  

```
(stream<tuple<rstring query> inTuple, sqlStatusType error> Import_3_out0) as
        Import_3 = Import()
    {
        param
            subscription : kind == "JDBC.NOTIFICATION" ;
    }
```

The import operator is directly connected to an error port in the  JDBCRun via export.  

I converted the error attribute to Rstring and got the following:
{errorCode=0,sqlState=""}

Looks like both the errorCode and sqlState are empty.  Please check code to see if it is properly initialized.

**Comment by chanskw** [comment]
I looked at the related code:

I think the errorCode and sqlState should be passed into the submitErrorTuple function.  Otherwise, we could get into the situation when we are submitting the error tuple, the errorCode and sqlState are overwritten in the process method, when there are multiple streams feeding into the input port?

catch (SQLException e){                
                 errorCode = e.getErrorCode();
        	sqlState = e.getSQLState();
        	TRACE.log(TraceLevel.DEBUG, "SQL Exception Error Code: " + errorCode);
        	TRACE.log(TraceLevel.DEBUG, "SQL EXCEPTION SQL State: " + sqlState);
        	if (hasErrorPort){
        		// submit error message
        		submitErrorTuple(errorOutputPort, tuple);
        	}

**Comment by chanskw** [comment]
Here's something you can use to simulate a connection lost.

In JDBCRun.allPortsReady()

I created a thread.  The thread sleeps for x number of seconds.  When the timer pops, we disconnect from the db.

In JDBCConnectionHelper, I added a method to force the connection to be closed.  This method simply call connection.close()...  I purposefully did not change any state or do any checks to simulate something unexpected. Call this method on the thread from allPortsReady.

**Comment by yiminy** [comment]
Just delivered a fix on this issue. Please let me know if it does not work for you.
Meanwhile, as existing design on sqlStatus and error tuple processing are not very clear, I am going to re-design this feature.

**Comment by chanskw** [comment]
Thanks!  let me try that out.

**Comment by yiminy** [comment]
The operator will check if the JDBC connection is valid before processing the tuple. If it is not valid, the operator will reset the connection. 
So there are two scenarios:
1. The connection lost before tuple processing, you will not get the error tuple from error port.
2. The connection lost during tuple processing, you will get the error tuple from error port.

**Comment by anouri** [comment]
corrected in JDBC version 1.4.0
https://github.com/IBMStreams/streamsx.jdbc/releases/tag/v1.4.0
